components/nexus-repository/src/main/java/org/sonatype/nexus/repository/config/UniqueRepositoryNameValidator.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2008-2015 Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.repository.config;
+
+import java.util.Locale;
+
+import javax.inject.Named;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+import org.sonatype.nexus.repository.Repository;
+import org.sonatype.nexus.repository.manager.RepositoryManager;
+
+import com.google.inject.Inject;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Ensure that Repository names are unique case-insensitively.
+ *
+ * @since 3.0
+ */
+@Named
+public class UniqueRepositoryNameValidator
+    implements ConstraintValidator<UniqueRepositoryName, String>
+{
+  private final RepositoryManager repositoryManager;
+
+  @Inject
+  public UniqueRepositoryNameValidator(final RepositoryManager repositoryManager) {
+    this.repositoryManager = checkNotNull(repositoryManager);
+  }
+
+  @Override
+  public void initialize(final UniqueRepositoryName constraintAnnotation) {
+    // do nothing
+  }
+
+  @Override
+  public boolean isValid(final String value, final ConstraintValidatorContext context) {
+    for (Repository repository : repositoryManager.browse()) {
+      if (repository.getName().toLowerCase(Locale.ENGLISH).equals(value.toLowerCase(Locale.ENGLISH))) {
```

Comment:
After I saw this I was considering adding Strings2.equalsIgnoreCase(String,String) as well, since Strings.equalsIgnoreCase() doesn't have a locale option... fucking locale